homeassistant/components/diagnostics/__init__.py

```diff
@@ -121,17 +123,44 @@ def handle_get(
     )
 
 
-def _get_json_file_response(
+async def _async_get_json_file_response(
+    hass: HomeAssistant,
     data: dict | list,
     filename: str,
+    domain: str,
     d_type: DiagnosticsType,
     d_id: str,
     sub_type: DiagnosticsSubType | None = None,
     sub_id: str | None = None,
 ) -> web.Response:
     """Return JSON file from dictionary."""
+    hass_sys_info = await async_get_system_info(hass)
+
+    integration: Integration = hass.data[DATA_INTEGRATIONS][domain]
+
+    custom_components = {}
+    cc_domain: str
+    cc_obj: Integration
+    for cc_domain, cc_obj in hass.data[DATA_CUSTOM_COMPONENTS].items():
+        custom_components[cc_domain] = {
+            "version": cc_obj.manifest["version"],
+            "requirements": [
+                req
+                for req in cc_obj.manifest["requirements"]
+                if "://" not in req  # ignore url based requirements
+            ],
+        }
     try:
-        json_data = json.dumps(data, indent=2, cls=ExtendedJSONEncoder)
+        json_data = json.dumps(
+            {
+                "home-assistant": hass_sys_info,
```

Comment:
I think we need to make use of this more explicitly, as it contains the username, which I think we should not add as the default output (this result is now used directly).

Reply:
I think the goal is to know if it is run as root or not? If so, we should add a flag for that (instead of putting in the name).

Reply:
for now `user` is just redacted, but if it is just about to know, if it is `root` or not, than i will change it this way?

Reply:
It only matter if root or not

Reply:
would this be ok?
```python
    hass_sys_info = await async_get_system_info(hass)
    if run_as := hass_sys_info.get("user") is not None:
        if run_as == "root":
            hass_sys_info["run_as_root"] = true
        else:
            hass_sys_info["run_as_root"] = false
        hass_sys_info.update({"user": REDACTED})
```

Reply:
I would:

```py
hass_sys_info = await async_get_system_info(hass)
hass_sys_info["run_as_root"] = hass_sys_info["user"] == "root"
del hass_sys_info["user"]
```

Reply:
Or (without knowing the impact it has at this point), change `async_get_system_info`

Reply:
Since it can be used by custom integration, I don't think we should remove the `user` key from `async_get_system_info`, we could make it `None` during a deprecation period, but changing that should not be done in this PR.

<https://github.com/home-assistant/core/pull/64670#discussion_r790148992> would work here :+1:

Reply:
> I don't think we should remove the user key from async_get_system_info, we could make it None during a deprecation period

Oh, sorry, I should have been more clear. I meant adding `runs_as_root` or something to the `async_get_system_info`.